File: swift/problem551.swift
class Solution {
    func checkRecord(_ s: String) -> Bool {
        var flag_A = false
        var flag_L = false
        var flag_LL = false
        
        for char in s {
            if (char == "A") {
                if flag_A {
                    return false
                } else {
                    flag_A = true
                }
                flag_L = false
                flag_LL = false
            } else if (char == "L") {
                if flag_LL {
                    return false
                } else if flag_L {
                    flag_LL = true
                } else {
                    flag_L = true
                }
            } else {
                flag_L = false
                flag_LL = false
            }
        }
        
        return true
    }
}

File: cpp/problem551.cpp
#include "mystd.h"
using namespace std;

/**
 * pay attention to the 'continuous' condition of L
 */
class Solution {
public:
    bool checkRecord(string s) {
        int ct_A = 0, ct_L = 0;
        for(auto c: s){
            ct_A += c=='A';
            ct_L = (c=='L')?(ct_L+1):0;
            if( ct_A > 1 || ct_L > 2 ) return false;
        }
        return true;
    }
};

int main(int argc, char const *argv[]){
    Solution sol;
    string s1 = "PPALLP", s2 = "PPALLL", s3 = "LALL";

    cout<< sol.checkRecord(s1) << '\t'
        << sol.checkRecord(s2) << '\t'
        << sol.checkRecord(s3) << endl;
    return 0;
}
